secrets/secrets.go

```diff
@@ -298,3 +324,45 @@ func NewSecrets(r io.Reader) (*Secrets, error) {
 	}
 	return secrets, nil
 }
+
+func csiPathParser(path string, inputSecrets Document) (secretsDocument Document, err error) {
+	secretsDocument = Document{
+		Secrets: make(map[string]GenericSecret),
+	}
+	file, err := os.Open(path)
+	if err != nil {
+		return secretsDocument, err
+	}
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return secretsDocument, err
+	}
+	if fileInfo.IsDir() {
+		files, err := ioutil.ReadDir(path)
```

Comment:
oh so here is where `ioutil` is used. for this you want to use [`os.ReadDir`](https://pkg.go.dev/os#ReadDir) instead.